scripts/convert_lit_checkpoint.py

```diff
@@ -0,0 +1,238 @@
+            from_name, number = layer_template(name, 2)
+            q = "model.layers.{}.self_attn.q_proj.weight".format(number)
+            k = "model.layers.{}.self_attn.k_proj.weight".format(number)
+            v = "model.layers.{}.self_attn.v_proj.weight".format(number)
+        else:
+            to_name = get_to_name(name, weight_map)
+
+        # handle param
+        if name.endswith(".attn.attn.weight"):
+            qkv = load_param(param)
+            qp, kp, vp = tensor_split(qkv, config, "llama")
+            for to_name, _param in zip((q, k, v), (qp, kp, vp)):
+                if saver is not None:
+                    param = saver.store_early(_param)
+                state_dict[to_name] = param
+        else:
+            param = load_param(param)
+            if saver is not None:
+                param = saver.store_early(param)
+            state_dict[to_name] = param
+
+
+def tensor_split(param: Union[torch.Tensor, NotYetLoadedTensor], config: Config, model_name: str) -> torch.Tensor:
+    if model_name != "llama":
+        raise NotImplementedError(f"{model_name}")
+    else:
+
+        def middle(start, head_size) -> int:
+            return start + head_size
+
+        def end(start, head_size) -> int:
+            return start + int(head_size * 2)
+
+        head_size = config.head_size
+        nobs = param.shape[0] + 1
+        nsplits = len(("q", "k", "v"))
+        splice_starts = range(head_size, nobs, head_size * nsplits)
+
+        splices = [(start, middle(start), end(start)) for start in splice_starts]
+
+    qc = ()
+    kc = ()
+    vc = ()
+
+    for splice in splices:
+        qs, ks, vs = splice
+        qc += (param[qs - head_size : qs, :],)
+        kc += (param[qs:ks, :],)
+        vc += (param[ks:vs, :],)
+
+    q = torch.cat(qc)
+    k = torch.cat(kc)
+    v = torch.cat(vc)
+
+    return q, k, v
+
+
+def get_to_name(lit_key_name: str, weight_map: Dict[str, str]) -> str:
+    for k, v in weight_map.items():
+        if lit_key_name == v:
+            return k
+
+
+@torch.inference_mode()
+def convert_lit_checkpoint(
+    *,
+    checkpoint_dir: Path = Path("checkpoints/tiiuae/falcon-7b"),
+    checkpoint_name: str = "lit_model_finetuned.pth",
+    model_name: Optional[str] = None,
+) -> None:
+    if model_name is None:
+        model_name = checkpoint_dir.name
+    config = Config.from_name(model_name)
+
+    if "falcon" in model_name:
+        copy_fn = partial(copy_weights_falcon, "40b" if config.n_embd == 8192 else "7b")
+    else:
+        raise NotImplementedError(f"Conversion for {model_name} is not yet supported")
+
+    # initialize a new empty state dict to hold our new weights
+    sd = {}
+
+    # checkpoint_name cannot be hardcoded because there exists different outputs such as
+    # ("lit_model_finetuned.pth", "lit_model_lora_finetuned.pth", "lit_model_adapter_finetuned.pth"")
+    pth_file = checkpoint_dir / checkpoint_name
+    bin_file = checkpoint_name.replace(".pth", ".bin")
```

Comment:
Why is this replacement necessary?

Reply:
I noticed all HF models are bin and not pth; so I converted just in case HF requires .bin file ext. If they do not, then the replacement is not needed. 

The replacement, though not really necessary, could be useful to distinguish model formats in addition to `lit_model` for our format.

Reply:
changed the convention in [16c9675](https://github.com/Lightning-AI/lit-gpt/pull/273/commits/16c96750923a820b384c63bed672c49aa60e6b50)

users will now be prompted in the tutorial to pass the checkpoint_name without a file extension, and .bin will be joined to the given checkpoint_name